src/framegrab/grabber.py

```diff
@@ -1035,6 +1045,50 @@ def _apply_camera_specific_options(self, options: dict) -> None:
             pass
 
 
+class RaspberryPiCSI2FrameGrabber(FrameGrabber):
+    "For CSI2 cameras connected to Raspberry Pis through their dedicated camera port"
+
+    def __init__(self, config: dict):
+        self.config = config
+
+        # This will also detect USB cameras, but according to the documentation CSI2
+        # cameras attached to the dedicated camera port will always come before USB
+        # cameras in the resulting list of camera dictionaries
+        # https://datasheets.raspberrypi.com/camera/picamera2-manual.pdf#page=66.21
+        cameras = Picamera2.global_camera_info()
+
+        if not cameras:
+            raise ValueError("No CSI2 cameras were found. Is your camera connected?")
+
+        # Since global_camera_info() also finds USB cameras, we will only use the first
+        # entry. USB cameras must be found through their specific FrameGrabber. Note
+        # that only a single CSI2 camera is supported at this time.
+        picam2 = Picamera2()
+        picam2.configure(picam2.create_still_configuration())
+        picam2.start()
+        logger.info(f"Connected to Raspberry Pi CSI2 camera with id {(cameras[0])['Id']}")
+
+        self.camera = picam2
+
+    def _grab_implementation(self) -> np.ndarray:
+        frame = self.camera.capture_array()
+
+        # Convert to BGR for opencv
+        frame = cv2.cvtColor(np.asanyarray(frame), cv2.COLOR_BGR2RGB)
+
+        return frame
+
+    def _apply_camera_specific_options(self, options: dict) -> None:
```

Comment:
Interesting that resolution cannot be set for this camera type. The picamera module really doesn't provide anything for that? When I used Raspberry Pi camera v2 in the past, I was able to change the resolution, however I did it through OpenCV.

Reply:
You're right, it can be changed. I was shooting for the minimum working version of getting these cameras supported and missed that functionality, sorry about that. For now, since we're in a bit of a rush, I'll change the message to something more specific to framegrab not supporting setting resolution for CSI2 cameras yet, and file a jira ticket to add that functionality later (along with autograb support and multiple CSI2 cameras).